packages/core/src/lib/toast/toast-banner.svelte

```diff
@@ -0,0 +1,60 @@
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { Button, Icon, type IconName, type ToastVariantType } from '$lib';
+import { ToastVariant } from './variants';
+
+/** The type of toast banner to be displayed to the user */
+export let variant: ToastVariantType;
+/** Function called when the close button is clicked */
+export let dismiss: () => void;
+
+/** Additional CSS classes to pass to the banner. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+let icon: IconName | null = null;
+let iconClasses = '';
+
+$: {
+  // TODO: use a `switch` statement when extending the variants
+  if (variant === ToastVariant.Success) {
+    icon = 'check-circle';
+    iconClasses = 'text-success-dark';
+  }
+}
```

Comment:
This logic looks like it could live in `variants.ts` and be a little easier to read